web_copy_confirm/static/src/js/web_copy_confirm.js

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018 DynApps <http://www.dynapps.be>
+// @author Stefan Rijnhart <stefan@opener.amsterdam>
+// @author Robin Conjour <rconjour@demolium.com>
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+odoo.define("web_copy_confirm.web_copy_confirm", function (require) {
+    "use strict";
+
+    const Core = require("web.core");
+    const FormController = require("web.FormController");
+    const Dialog = require("web.Dialog");
+    const _t = Core._t;
+
+    FormController.include({
+        init: function () {
+            this._super.apply(this, arguments);
+        },
+        /**
+         * Trigger the confirmation dialog when duplicating records
+         * @returns {Promise<void>}
+         * @private
+         */
+        _onDuplicateRecordConfirm: async function () {
+            Dialog.confirm(
+                this,
+                _t("Are you sure that you would like to copy this record?"),
+                {
+                    confirm_callback: () => this._onDuplicateRecord(),
+                }
+            );
+        },
+        /**
+         * Override the "duplicate" in the action menu
+         * @returns {any}
+         * @private
+         */
+        _getActionMenuItems: function () {
+            const props = this._super(...arguments);
+
+            if (props && props.items && props.items.other) {
+                const other_list = props.items.other;
+                const duplicate_index = other_list.findIndex(
+                    (item) =>
+                        item.callback.toString() ===
+                        "() => this._onDuplicateRecord(this)"
+                );
+                other_list[duplicate_index].callback = () =>
+                    this._onDuplicateRecordConfirm();
+
+                return Object.assign(props, {
```

Comment:
You are modifiying `props.items.other` inline so you don't have to reassign it to the toolbar actions object.
![image](https://user-images.githubusercontent.com/1033124/104919327-5d498d80-5996-11eb-8b1e-7d6c4d533c65.png)

Reply:
Oh, and because of the indentation level, you are not returning the return value of `super` if your condition does not come out as true.

Reply:
Changed!